src/app.js

```diff
@@ -1,21 +1,35 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const http = require('http');
+const fs = require('fs');
+
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer((req, res) => {
+  const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+  const pathnameArray = normalizedUrl.pathname
+    .split('/')
+    .filter(path => path !== '');
+
+  if (pathnameArray[0] !== 'file') {
+    res.end('Correct request is: "/file/<path_to_file>"');
+
+    return;
+  }
+
+  const filename = pathnameArray.slice(1).join('/') || 'index.html';
+
+  fs.readFile(`./src/public/${filename}`, (err, data) => {
+    if (!err) {
+      res.end(data);
+    }
+
+    res.statusCode = 404;
```

Comment:
The statusCode 404 and ending the response should be inside the error block of fs.readFile. If the file is successfully read, the response will be ended prematurely before the data is sent.

Reply:
If there is no error, response will be ended with data in line 24. But if there is some error, "if" block will be skipped and response will be ended with status code 404. So it's working correctly